src/m365/search/commands/externalconnection/externalconnection-get.ts

```diff
@@ -0,0 +1,89 @@
+import { Logger } from '../../../../cli';
+import { ExternalConnectors } from '@microsoft/microsoft-graph-types';
+import GraphCommand from '../../../base/GraphCommand';
+import GlobalOptions from '../../../../GlobalOptions';
+import commands from '../../commands';
+import { CommandOption } from '../../../../Command';
+import request from '../../../../request';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  name?: string;
+}
+
+class SearchExternalConnectionGetCommand extends GraphCommand {
+
+  public get name(): string {
+    return commands.EXTERNALCONNECTION_GET;
+  }
+
+  public get description(): string {
+    return 'Retrieves the specified external connections';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    let endpoint: string = `${this.resource}/v1.0/external/connections`;
+    
+    if (args.options.id) {
+      endpoint += `/${encodeURIComponent(args.options.id)}`;
+    }
+    else {
+      endpoint += `?$filter=name eq '${encodeURIComponent(args.options.name as string)}'`;
+    }
+
+    const requestOptions: any = {
+      url: endpoint,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    request.get<ExternalConnectors.ExternalConnection>(requestOptions)
+      .then((res: ExternalConnectors.ExternalConnection): void => {
+        logger.log(res);
```

Comment:
We need to adjust this, because right now it returns data in two different shapes: if you refer to the connection by its ID, it will return the object with the information about the connection. If you however refer to the connection by its name, it will return an array with the matched connection object. In both cases we should return just the matched object. We should also ensure that we handle errors the same way: no matter if the user specifies the ID or the name, they should get a similar error message.